Composer/packages/client/src/pages/design/createDialogModal.tsx

```diff
@@ -32,47 +33,40 @@ export const CreateDialogModal: React.FC<CreateDialogModalProps> = (props) => {
   const { state } = useContext(StoreContext);
   const { dialogs } = state;
   const { onSubmit, onDismiss, isOpen } = props;
-  const initialFormData: DialogFormData = { name: '', description: '' };
-  const [formData, setFormData] = useState(initialFormData);
-  const [formDataErrors, setFormDataErrors] = useState<{ name?: string }>({});
+  const formConfig: FieldConfig<DialogFormData> = {
+    name: {
+      required: true,
+      validate: (value) => {
+        const nameRegex = /^[a-zA-Z0-9-_.]+$/;
```

Comment:
can we move this constant outside at the module level scope rather than function level

Reply:
I think since this is the only place that `nameRegex` is used, there is no need to promote to a module constant. I actually find that to be more of a cognitive load (needing to scan the file to see where the const is defined, etc). If this were used in multiple places in this module, moving it would be appropriate.